CONTRIBUTING.md

```diff
@@ -325,6 +326,84 @@ if you change the history (rebase, squash, amend), and use git push --force to
+wrappers for Markdown and Syntax Highlighting parsers, session handling, etc.
+Helpers are located in `app/helpers.rb` If you add a helper, update the list of existing helpers in this file (the convention is `ClassName: 'filename'`).
+
+**Presenters** are for showing information that would be beneficial to users or the views but maybe not the best way to store data.
+For example, in exercism, most times are stored in UTC (generally most servers store time this way) but... users might want to see times in their own timezone.
+Presenters can take relevant information (like showing comment notifications on the dashboard) and transform it to be more personal.
+Presenters are located in `app/presenters.rb`. If you add a presenter, update the list of existing presenters in this file (the convention is `ClassName: 'filename'`).
+
+**Routes** in Sinatra are kind of like a combination of routes and controllers in Rails.
+In a route, you can specify endpoints (like `exercism.io/login`) and define how the app should respond (i.e. accept some credentials and log the user in or handle errors accordingly).
+Routes are located in `app/routes.rb`. If you add a route, update the list of existing routes in this file (the convention is `ClassName: 'filename'`).
+Routes are unique in that you'll also have to update the main `app.rb` with the new route info (the convention is `Routes::ClassName`).
+
+**Views** are a collection of templates that create html.
+Templates are an awesome way to have embedded Ruby in your views so that you can serve up information,
+iterate over items without repeating yourself (like creating tables of data), etc.
+The templating engine that exercism uses is [ERB](https://en.wikipedia.org/wiki/ERuby).
+
+#### Client and Frontend (JavaScript, Custom Directives, Bootstrap, etc)
+The client side JS is a bit... spread out. There are two different JS related directories: `client/` and `frontend/`.
+Both have angular elements but `client/` is in JS and `frontend/` is mostly CoffeeScript.
+
+**Frontend** is where Exercism stores all of it's production client side code.
+Exercism doesn't have much JS (outside of bootstrap) but `frontend/` does handle comment threads on submissions, markdown preview on comments, submission code (expanding and contracting), etc. If you suspect that there is frontend wizardry happening and you haven't seen it in bootstrap, most likely it's in here.
+
+**Client** contains the code for the `comment-box` angular directive (custom html tag). Exercism doesn't use `comment-box` so this section might go away...
+
+#### DB (Migrations)
+To create a migration you can use `rake db:generate:migration name=the_name_of_your_migration`.
+That will create a new migration in `db/migrate`.
+For help, look at the other migrations or at docs for ActiveRecord migrations.
+If you add, remove or rename database columns, you will most likely have to update the seeds as well.
```

Comment:
s/will most likely/may/ -- let's assume that they won't have to and only give them a heads up that they might.